# **NUMBERS PROGRAM**

# _**1. PRIME NUMBER PROGRAM :**_
### The given number should be only divisible by 1 and that number itself
### Example : 
### number=7
### 7%1==0 True - count=1
### 7%2==0 False
### 7%3==0 False
### 7%4==0 False
### 7%5==0 False
### 7%6==0 False
### 7%7==0 True - count=1+1=2
### count==2
### 2==2

### **1.1 CHECK NUMBER IS PRIME NUMBER OR NOT**

In [1]:
num=int(input())
start=1
count=0
while start<=num:
    if num%start==0:
        count+=1
    start+=1
if count==2:
    print('Prime Number')
else:
    print('Not a Prime Number')

 7


Prime Number


### **1.2 PRINT THE PRIME NUMBERS FROM M TO N**

In [3]:
m=int(input('Enter the Start Number :'))
n=int(input('Enter the End Number :'))
print(f'Prime Numbers from {m} to {n} is :')
while m<=n:
    num=m
    start=1
    count=0
    while start<=num:
        if num%start==0:
            count+=1
        start+=1
    if count==2:
        print(num,end=' | ')
    m+=1

Enter the Start Number : 1
Enter the End Number : 20


Prime Numbers from 1 to 20 is :
2 | 3 | 5 | 7 | 11 | 13 | 17 | 19 | 

### **1.3 PRINT THE FIRST N PRIME NUMBERS**

In [3]:
num=1
c=0
n=int(input('Enter the First N Number :'))
print(f'First {n} Prime Numbers is :')
while c<n:
    num1=num
    start=1
    count=0
    while start<=num1:
        if num1%start==0:
            count+=1
        start+=1
    if count==2:
        print(num1,end=' | ')
        c+=1
    num+=1

Enter the First N Number : 10


First 10 Prime Numbers is :
2 | 3 | 5 | 7 | 11 | 13 | 17 | 19 | 23 | 29 | 

# **2. FACTORIAL NUMBER PROGRAM :**
### The given number should be product of the number from that number to till it reaches to 1 ( n * (n-1) )
### Example :
### number=5 
### 5\*(5-1) = 5\*4 = 20
### 5\*(5-1)\*(4-1) = 20\*3 = 60
### 5\*(5-1)\*(4-1)\*(3-1) = 20\*3 = 60\*2 = 120
### 5\*(5-1)\*(4-1)\*(3-1)\*(2-1) = 20\*3 = 60\*2 = 120\*1 = 120

### **2.1 PRINT THE FACTORIAL OF THE NUMBER**

In [5]:
num=int(input())
fact=1
start=1
while start<=num:
    fact*=start
    start+=1
print(fact)

 5


120


### **2.2 PRINT THE FACTORIAL OF THE NUMBER FROM M TO N**

In [11]:
m=int(input('Enter the Start Number :'))
n=int(input('Enter the First N Number :'))
print(f"Factorial of Numbers from {m} to {n} is :")
while m<=n:
    num=m
    fact=1
    start=1
    while start<=num:
        fact*=start
        start+=1
    print(fact,end=' | ')
    m+=1

Enter the Start Number : 1
Enter the First N Number : 10


Factorial of Numbers from 1 to 10 is :
1 | 2 | 6 | 24 | 120 | 720 | 5040 | 40320 | 362880 | 3628800 | 

### **2.3 PRINT THE FIRST N FACTORIAL OF THE NUMBERS**

In [12]:
num=1
n=int(input('Enter the First N Number :'))
c=0
print(f"Factorial of Numbers from {m} to {n} is :")
while c<n:
    num1=num
    fact=1
    start=1
    while start<=num1:
        fact*=start
        start+=1
    print(fact,end=' | ')
    c+=1
    num+=1

Enter the First N Number : 10


Factorial of Numbers from 11 to 10 is :
1 | 2 | 6 | 24 | 120 | 720 | 5040 | 40320 | 362880 | 3628800 | 

# **3. FIBONACCI NUMBER PROGRAM :**
### The given number should be sum of the numbers upto nth term and that start from 0 & 1
### Example : 
### nthTerm = 6
### 0+1=1
### 1+1=2
### 1+2=3
### 2+3=5
### 3+5=8
### 5+8=13

### **3.1 PRINT THE FIBONACCI OF THE NUMBER**

In [25]:
nthTerm=int(input('Enter the How Many Nth Term :'))
print('FIBONACCI Numbers :')
n1=0
n2=1
count=0
sum=0
while count<nthTerm:
    print(n1,end=' | ')
    sum=sum+n1
    nth=n1+n2
    n1=n2
    n2=nth
    count+=1
print('\nSum :',sum)

Enter the How Many Nth Term : 10


FIBONACCI Numbers :
0 | 1 | 1 | 2 | 3 | 5 | 8 | 13 | 21 | 34 | 
Sum : 88


### **3.2 PRINT THE FIBONACCI NUMBER FROM M TO N** 

In [29]:
m=int(input('Enter the Start Number :'))
n=int(input('Enter the End Number :'))
print('FIBONACCI Numbers :')
while m<=n:
    nthTerm=m
    n1=0
    n2=1
    count=0
    while count<nthTerm:
        print(n1,end=' | ')
        nth=n1+n2
        n1=n2
        n2=nth
        count+=1
    print('')
    m+=1

Enter the Start Number : 1
Enter the End Number : 10


FIBONACCI Numbers :
0 | 
0 | 1 | 
0 | 1 | 1 | 
0 | 1 | 1 | 2 | 
0 | 1 | 1 | 2 | 3 | 
0 | 1 | 1 | 2 | 3 | 5 | 
0 | 1 | 1 | 2 | 3 | 5 | 8 | 
0 | 1 | 1 | 2 | 3 | 5 | 8 | 13 | 
0 | 1 | 1 | 2 | 3 | 5 | 8 | 13 | 21 | 
0 | 1 | 1 | 2 | 3 | 5 | 8 | 13 | 21 | 34 | 


### **3.3 PRINT THE FIRST N FIBONACCI NUMBERS**

In [32]:
num=1
n=int(input('Enter the First N Number :'))
print('FIBONACCI Numbers :')
c=0
while c<n:
    nthTerm=num
    n1=0
    n2=1
    count=0
    while count<nthTerm:
        print(n1,end=' | ')
        nth=n1+n2
        n1=n2
        n2=nth
        count+=1
    c+=1
    num+=1
    print('')

Enter the First N Number : 10


FIBONACCI Numbers :
0 | 
0 | 1 | 
0 | 1 | 1 | 
0 | 1 | 1 | 2 | 
0 | 1 | 1 | 2 | 3 | 
0 | 1 | 1 | 2 | 3 | 5 | 
0 | 1 | 1 | 2 | 3 | 5 | 8 | 
0 | 1 | 1 | 2 | 3 | 5 | 8 | 13 | 
0 | 1 | 1 | 2 | 3 | 5 | 8 | 13 | 21 | 
0 | 1 | 1 | 2 | 3 | 5 | 8 | 13 | 21 | 34 | 


# **4. NATURAL NUMBER PROGRAM :**
### Natural Numbers such as 1,2,3,4,5....n

### **4.1 PRINT THE SUM OF THE NATURAL NUMBERS**

In [8]:
num=int(input('Enter the Number :'))
print('Natural Numbers are :')
start=1
sum=0
while start<=num:
    print(start,end=' | ')
    sum+=start
    start+=1
print('\nSum :',sum)

Enter the Number : 10


Natural Numbers are :
1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 
Sum : 55


### **4.2 PRINT THE NATURAL NUMBERS FROM M TO N**

In [9]:
m=int(input('Enter the Start Number :'))
n=int(input('Enter the End Number :'))
print('Natural Numbers are :')
while m<=n:
    num=m
    start=1
    sum=0
    while start<=num:
        print(start,end=' | ')
        sum+=start
        start+=1
    print('\nSum :',sum)
    m+=1

Enter the Start Number : 1
Enter the End Number : 10


Natural Numbers are :
1 | 
Sum : 1
1 | 2 | 
Sum : 3
1 | 2 | 3 | 
Sum : 6
1 | 2 | 3 | 4 | 
Sum : 10
1 | 2 | 3 | 4 | 5 | 
Sum : 15
1 | 2 | 3 | 4 | 5 | 6 | 
Sum : 21
1 | 2 | 3 | 4 | 5 | 6 | 7 | 
Sum : 28
1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 
Sum : 36
1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 
Sum : 45
1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 
Sum : 55


### **4.3 PRINT THE SUM OF THE FIRST N NATURAL NUMBERS**

In [13]:
num=1
n=int(input('Enter the First N Number :'))
print('Natural Numbers are :')
count=0
while count<n:
    num1=num
    start=1
    sum=0
    while start<=num1:
        print(start,end=' | ')
        sum+=start
        start+=1
    count+=1
    print('\nSum :',sum)
    num+=1

Enter the First N Number : 10


Natural Numbers are :
1 | 
Sum : 1
1 | 2 | 
Sum : 3
1 | 2 | 3 | 
Sum : 6
1 | 2 | 3 | 4 | 
Sum : 10
1 | 2 | 3 | 4 | 5 | 
Sum : 15
1 | 2 | 3 | 4 | 5 | 6 | 
Sum : 21
1 | 2 | 3 | 4 | 5 | 6 | 7 | 
Sum : 28
1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 
Sum : 36
1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 
Sum : 45
1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 
Sum : 55


# **5. SPY NUMBER PROGRAM :**
### **Sum** of the Digits of a Given Number == **Product** of the Digits of a Given Number
### Example : 
### num=123
### sum of digitss -> 1+2+3=6
### product of the digits -> == 1\*2\*3=6
### sum == prod
### 6 == 6

### **5.1 CHECK THE NUMBER IS SPY NUMBER OR NOT**

In [14]:
num=int(input())
sum=0
prod=1
while num!=0:
    d=num%10
    sum+=d
    prod*=d
    num//=10
if sum==prod:
    print('SPY NUMBER')
else:
    print('Not a Spy Number')

 123


SPY NUMBER


### **5.2 PRINT THE SPY NUMBERS FROM M TO N**

In [18]:
m=int(input('Enter the start Number :'))
n=int(input('Enter the end Number :'))
print(f'SPY NUMBERS from {m} to {n} is :')
while m<=n:
    num=m
    sum=0
    prod=1
    while num!=0:
        d=num%10
        sum+=d
        prod*=d
        num//=10
    if sum==prod:
        print(m,end=' | ')
    m+=1

Enter the start Number : 1
Enter the end Number : 200


SPY NUMBERS from 1 to 200 is :
1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 22 | 123 | 132 | 

### **5.3 PRINT THE FIRST N SPY NUMBERS**

In [20]:
num=1
n=int(input('Enter the First N Number :'))
count=0
print(f'SPY NUMBERS from {m} to {n} is :')
while count<n:
    num1=num
    sum=0
    prod=1
    while num1!=0:
        d=num1%10
        sum+=d
        prod*=d
        num1//=10
    if sum==prod:
        print(num,end=' | ')
        count+=1
    num+=1

Enter the First N Number : 15


SPY NUMBERS from 201 to 15 is :
1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 22 | 123 | 132 | 213 | 231 | 312 | 

# **6. PERFECT NUMBER PROGRAM :**
### **Sum** of the Divisors(Except that number) == Given Number
### Example : 
### num=6 
### 6%1==0 True - sum=1
### 6%2==0 True - sum=1+2=3
### 6%3==0 True - sum=3+3=6
### 6%4==0 False
### 6%5==0 False
### num==sum 
### 6==6

### **6.1 CHECK THE NUMBER IS PERFECT NUMBER OR NOT**

In [6]:
num=int(input())
start=1
sum=0
while start<num:
    if num%start==0:
        sum+=start
    start+=1
if num==sum:
    print('Perfect')
else:
    print('Not a Perfect')

 6


Perfect


### **6.2 PRINT THE PERFECT NUMBERS FROM M TO N**

In [9]:
m=int(input('Enter the start number :'))
n=int(input('Enter the End number :'))
print('PERFECT NUMBERS ARE :')
while m<=n:
    num=m
    start=1
    sum=0
    while start<num:
        if num%start==0:
            sum+=start
        start+=1
    if num==sum:
        print(m,end=' | ')
    m+=1

Enter the start number : 1
Enter the End number : 1000


PERFECT NUMBERS ARE :
6 | 28 | 496 | 

### **6.3 PRINT THE FIRST N PRIME NUMBERS**

In [1]:
num=1
n=int(input('Enter the First N number :'))
print('PERFECT NUMBERS ARE :')
count=0
while count<n:
    num1=num
    start=1
    sum=0
    while start<num1:
        if num1%start==0:
            sum+=start
        start+=1
    if num==sum:
        print(num1,end=' | ')
        count+=1
    num+=1

Enter the First N number : 3


PERFECT NUMBERS ARE :
6 | 28 | 496 | 

# **7. NEON NUMBER PROGRAM :**
### **Sum** of **Square** of the digits Number == Given Number
### Example :
### num = 9
### square=> 9 * 9 = 81
### for 81 take sum of the digits, i.e, 8 + 1 =9
### Sum of Square of Digit == given Number
### 9 == 9

### **7.1 CHECK THE NUMBER IS NEON OR NOT**

In [7]:
num=int(input())
sqr=num**2
n=num
sum=0
while sqr!=0:
    d=sqr%10
    sum+=d
    sqr//=10
if sum==n:
    print('Neon Number')
else:
    print('Not a Neon Number')

 9


Neon Number


### **7.2 PRINT THE NEON NUMBERS FROM M TO N**

In [12]:
m=int(input('Enter the start Number :'))
n=int(input('Enter the End Number :'))
print('NEON NUMBERS :')
while m<=n:
    num=m
    sqr=num**2
    sum=0
    while sqr!=0:
        d=sqr%10
        sum+=d
        sqr//=10
    if sum==m:
        print(m,end=' | ')
    m+=1

Enter the start Number : 1
Enter the End Number : 500


NEON NUMBERS :
1 | 9 | 

### **7.3 PRINT THE FIRST N NUMBER OF NEON NUMBERS**

In [1]:
num=1
n=int(input('Enter the First N Number :'))
print('NEON NUMBERS :')
c=0
while c<n:
    num1=num
    sqr=num1**2
    sum=0
    while sqr!=0:
        d=sqr%10
        sum+=d
        sqr//=10
    if sum==num:
        print(num,end=' | ')
        c+=1
    num+=1

Enter the First N Number : 2


NEON NUMBERS :
1 | 9 | 

# **8. STRONG NUMBER PROGRAM :**
### **Sum** of the Factorial(Product) of digits of a Given Number == Given Number
### Example : num = 145
### fact=1 - 1 - sum=1
### fact=4 - 4\*3\*2\*1 = 24 - sum=1+24=25
### fact=5 - 5\*4\*3\*2\*1 = 120 - sum=25+120=145
### sum of fact == given number
### 145 == 145

### **8.1 CHECK THE NUMBER IS STRONG NUMBER OR NOT**

In [7]:
num=int(input())
n=num
sum=0
while num!=0:
    d=num%10
    fact=1
    start=1
    while start<=d:
        fact*=start
        start+=1
    sum+=fact
    num//=10
if sum==n:
    print('Strong Number')
else:
    print('Not a Strong Number')

 145


Strong Number


### **8.2 PRINT THE STRONG NUMBERS FROM M TO N**

In [12]:
m=int(input('Enter the Start Number :'))
n=int(input('Enter the End Number :'))
print('STRONG NUMBER :')
while m<=n:
    num=m
    sum=0
    while num!=0:
        d=num%10
        fact=1
        start=1
        while start<=d:
            fact*=start
            start+=1
        sum+=fact
        num//=10
    if sum==m:
        print(m,end=' | ')
    m+=1

Enter the Start Number : 1
Enter the End Number : 500


STRONG NUMBER :
1 | 2 | 145 | 

### **8.3 PRINT THE FIRST N STRONG NUMBERS**

In [13]:
m=1
n=int(input('Enter the First N Number :'))
print('STRONG NUMBER :')
c=0
while c<n:
    num=m
    sum=0
    while num!=0:
        d=num%10
        fact=1
        start=1
        while start<=d:
            fact*=start
            start+=1
        sum+=fact
        num//=10
    if sum==m:
        print(m,end=' | ')
        c+=1
    m+=1

Enter the First N Number : 4


STRONG NUMBER :
1 | 2 | 145 | 40585 | 

# **9. ARMSTRONG NUMBER PROGRAAM :**
### **Length** of the given Number is the **sum** of **power** of each digit == Given Number
### Example : num = 153
### 153 - lenght is = 3
### pow by 3 to the each digit of given number
### 1**3 = 1
### 5**3 = 125
### 3**3 = 27
### Sum of the power
### sum = 1+125+27 
### sum = 153
### num == sum
### 153 == 153

### **9.1 CHECK THE GIVEN NUMBER IS ARMSTRONG NUMBER OR NOT**

In [3]:
num=int(input())
count=0
n=num
n1=num
while num!=0:
    count+=1
    num//=10
sum=0
while n!=0:
    d=n%10
    pow=d**count
    n//=10
    sum+=pow
if n1==sum:
    print('Armstrong Number')
else:
    print('Not a Armstrong Number')

 153


Armstrong Number


### **9.2 PRINT THE ARMSTRONG NUMBERS FROM M TO N**

In [2]:
m=int(input('Enter the First Number :'))
n=int(input('Enter the End Number :'))
print('ARMSTRONG NUMBERS')
while m<=n:
    num=m
    count=0
    num1=m
    num3=m
    while num!=0:
        count+=1
        num//=10
    sum=0
    while num1!=0:
        d=num1%10
        pow=d**count
        num1//=10
        sum+=pow
    if num3==sum:
        print(num3,end=' | ')
    m+=1

Enter the First Number : 1
Enter the End Number : 500


ARMSTRONG NUMBERS
1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 153 | 370 | 371 | 407 | 

### **9.3 PRINT THE FIRST N ARMSTRONG NUMBERS**

In [3]:
m=1
n=int(input('Enter the First N Number :'))
print('ARMSTRONG NUMBERS')
c=0
while c<n:
    num=m
    count=0
    num1=m
    num3=m
    while num!=0:
        count+=1
        num//=10
    sum=0
    while num1!=0:
        d=num1%10
        pow=d**count
        num1//=10
        sum+=pow
    if num3==sum:
        print(num3,end=' | ')
        c+=1
    m+=1

Enter the First N Number : 12


ARMSTRONG NUMBERS
1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 153 | 370 | 371 | 

# **10. HAPPY NUMBER PROGRAM :**
### **Sum** of the **Square** digit of a given number == 1 (do repitedly until it becomes single digit , wehre 1 means Happy Number, 4 means unhappy or sad number
### Example : num = 19
### 1*\*2 + 9*\*2 = 1 + 81 = 82 - Single Digit - False
### 8*\*2 + 2*\*2 = 64 + 4 = 68 - Single Digit - False
### 6*\*2 + 8*\*2 = 36 + 64 = 100 - Single Digit - False
### 1*\*2 + 0*\*2 + 0*\*2 = 1 + 0 + 0 = 1 - Single Digit - True
### Sum of Square of digit == 1
### 1==1

### **10.1 CHECK THE NUMBER IS HAPPY NUMBER OR NOT**

In [2]:
num=int(input())
while num!=1 and num!=4:
    sum=0
    while num>0:
        d=num%10
        sum+=d**2
        num//=10
    num=sum
if num==1:
    print('Happy Number')
else:
    print('Not a Happy Number')

 44


Happy Number


### **10.2 PRINT ALL HAPPY NUMBERS FROM M TO N**

In [1]:
m=int(input('Enter the Start Number :'))
n=int(input('Enter the End Number :'))
print('HAPPY NUMBERS')
while m<=n:
    num=m
    while num!=1 and num!=4:
        sum=0
        while num>0:
            d=num%10
            sum+=d**2
            num//=10
        num=sum
    if num==1:
        print(m,end=' | ')
    m+=1

Enter the Start Number : 1
Enter the End Number : 100


HAPPY NUMBERS
1 | 7 | 10 | 13 | 19 | 23 | 28 | 31 | 32 | 44 | 49 | 68 | 70 | 79 | 82 | 86 | 91 | 94 | 97 | 100 | 

### **10.3 PRINT STARTING OF N HAPPY NUMBERS**

In [3]:
num=1
n=int(input('Enter the First N Number :'))
print('HAPPY NUMBERS')
c=0
while c<n:
    num1=num
    while num1!=1 and num1!=4:
        sum=0
        while num1>0:
            d=num1%10
            sum+=d**2
            num1//=10
        num1=sum
    if num1==1:
        print(num,end=' | ')
        c+=1
    num+=1

Enter the First N Number : 10


HAPPY NUMBERS
1 | 7 | 10 | 13 | 19 | 23 | 28 | 31 | 32 | 44 | 

# **11. DUCK NUMBER PROGRAM :**
### The Number Should contain atlease one **Zero** and **Zero** Should not be start number
### Example : num = 1234 & num = 0231 Is Not a Duck Number
#### 1203 , 4029, 4000000123 is Duck Number

### **11.1 CHECK NUMBER IS DUCK NUMBER OR NOT**

In [9]:
num=int(input())
zero_found=0
while num>0: 
    d=num%10
    if d==0:
        zero_found=1
    num//=10
if zero_found==1:
    print("It is a Duck Number")
else:
    print("Not a Duck Number")

 0123


Not a Duck Number


### **11.2 PRINT ALL THE DUCK NUMBER FROM M TO N**

In [15]:
m=int(input('Enter the Start Number :'))
n=int(input('Enter the End Number :'))
print('DUCK NUMBER')
while m<=n:
    num=m
    zero_found=0
    while num>0: 
        d=num%10
        if d==0:
            zero_found=1
        num//=10
    if zero_found==1:
        print(m,end=' | ')
    m+=1

Enter the Start Number : 1
Enter the End Number : 150


DUCK NUMBER
10 | 20 | 30 | 40 | 50 | 60 | 70 | 80 | 90 | 100 | 101 | 102 | 103 | 104 | 105 | 106 | 107 | 108 | 109 | 110 | 120 | 130 | 140 | 150 | 

### **11.3 PRINT ALL THE FIRST N DUCK NUMBERS**

In [14]:
m=1
n=int(input('Enter the First N Number :'))
print('DUCK NUMBERS')
c=0
while c<n:
    num=m
    zero_found=0
    while num>0: 
        d=num%10
        if d==0:
            zero_found=1
        num//=10
    if zero_found==1:
        print(m,end=' | ')
        c+=1
    m+=1

Enter the First N Number : 15


DUCK NUMBERS
10 | 20 | 30 | 40 | 50 | 60 | 70 | 80 | 90 | 100 | 101 | 102 | 103 | 104 | 105 | 

# **12. HARSHAD(NIVEN) NUMBER PROGRAM :**
### **Sum** of the digit of a given number  , and that number should be divisible by their **Sum**
### Example : num = 18
### sum = 1 + 8 = 9
### num%sum==0
### 18%9==0 True

### **12.1 CHECK GIVEN NUMBER IS HARSHAD NUMBER OR NOT**

In [5]:
num=int(input())
num1=num
sum=0
while num!=0:
    d=num%10
    sum+=d
    num//=10
if num1%sum==0:
    print('Harshad Number')
else:
    print('Not a Harshad Number')

 18


Harshad Number


### **12.2 PRINT ALL HARSHAD NUMBER FROM M TO N**

In [6]:
m=int(input('Enter the Start Number :'))
n=int(input('Enter the End Number :'))
while m<=n:
    num=m
    num1=num
    sum=0
    while num!=0:
        d=num%10
        sum+=d
        num//=10
    if num1%sum==0:
        print(m,end=' | ')
    m+=1

Enter the Start Number : 1
Enter the End Number : 50


1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 12 | 18 | 20 | 21 | 24 | 27 | 30 | 36 | 40 | 42 | 45 | 48 | 50 | 

### **12.3 PRINT ALL FIRST N HARSHAD NUMBER**

In [8]:
m=1
n=int(input('Enter the First N Number :'))
c=0
while c<n:
    num=m
    num1=num
    sum=0
    while num!=0:
        d=num%10
        sum+=d
        num//=10
    if num1%sum==0:
        print(m,end=' | ')
        c+=1
    m+=1

Enter the First N Number : 20


1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 12 | 18 | 20 | 21 | 24 | 27 | 30 | 36 | 40 | 42 | 

# **13. DISARIUM NUMBER PROGRAM :**
### Increase the power by 1 for each digit in a given number and **Sum** that Number == number
### Example : num = 175
### get digit and apply power by 1 and increase the power to nexr digit
### 1**1 = 1 
### 7**2 = 49
### 5**3 = 125
### Sum = 1 + 49 + 125 = 175
### Sum == Num
### 175 == 175

### **13.1 CHECK THAT GIVEN NUMBER IS DISARIUM NUMBER OR NOT**

In [16]:
num=int(input())
i=1
rev=0
num1=num
#reverse the number
while num!=0:
    d=num%10
    rev=rev*10+d
    num//=10
sum=0
while rev!=0:
    d=rev%10
    res=d**i
    sum+=res
    i+=1
    rev//=10
if sum==num1:
    print('Disarium Number')
else:
    print('Not a Disarium Number')

 89


Disarium Number


### **13.2 PRINT ALL DISARIUM NUMBERS FROM M TO N**

In [18]:
m=int(input('Enter the Start Number :'))
n=int(input('Enter the End Number :'))
print('DISARIUM NUMBER')
while m<=n:
    num=m
    i=1
    rev=0
    num1=num
    #reverse the number
    while num!=0:
        d=num%10
        rev=rev*10+d
        num//=10
    sum=0
    while rev!=0:
        d=rev%10
        res=d**i
        sum+=res
        i+=1
        rev//=10
    if sum==num1:
        print(m,end=' | ')
    m+=1

Enter the Start Number : 1
Enter the End Number : 200


DISARIUM NUMBER
1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 89 | 135 | 175 | 

### **13.3 PRINT FIRST N DISARIUM NUMBERS**

In [ ]:
m=1
n=int(input('Enter the First N Number :'))
print('DISARIUM NUMBER')
c=0
while c<n:
    num=m
    i=1
    rev=0
    num1=num
    #reverse the number
    while num!=0:
        d=num%10
        rev=rev*10+d
        num//=10
    sum=0
    while rev!=0:
        d=rev%10
        res=d**i
        sum+=res
        i+=1
        rev//=10
    if sum==num1:
        print(m,end=' | ')
        c+=1
    m+=1

Enter the First N Number : 20


DISARIUM NUMBER
1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 89 | 135 | 175 | 518 | 598 | 1306 | 1676 | 2427 | 2646798 | 

# **14. PRONIC NUMBER PROGRAM :**
### Product of two consecutive integres == given number
### Example : num = 20
### num*(num+1)
### Note : Number starts from 1
### 1\*(1+1) = 1\*2 = 2
### 2\*(2+1) = 2\*3 = 6
### 3\*(3+1) = 3\*4 = 12
### 4\*(4+1) = 4\*5 = 20 True
### Product of two consecutive integers == Number
### 20 == 20

### **14.1 CHECK GIVEN NUMBER IS PRONIC NUMBER OR NOT**

In [1]:
num=int(input())
start=1
flag=0
while start<=num:
    if start*(start+1)==num:
        flag=1
        break
    start+=1
if flag==1:
    print('Pronic Number')
else:
    print('Not a Pronic Number')

 42


Pronic Number


### **14.2 PRINT ALL PRONIC NUMBER FROM M TO N**

In [2]:
m=int(input('Enter the Start Number :'))
n=int(input('Enter the End Number :'))
while m<=n:
    num=m
    start=1
    flag=0
    while start<=num:
        if start*(start+1)==num:
            flag=1
            break
        start+=1
    if flag==1:
        print(m,end=' | ')
    m+=1

Enter the Start Number : 1
Enter the End Number : 200


2 | 6 | 12 | 20 | 30 | 42 | 56 | 72 | 90 | 110 | 132 | 156 | 182 | 

### **14.3 PRINT FIRST N PRONIC NUMBER**

In [3]:
m=1
n=int(input('Enter the First Number :'))
c=0
while c<n:
    num=m
    start=1
    flag=0
    while start<=num:
        if start*(start+1)==num:
            flag=1
            break
        start+=1
    if flag==1:
        print(m,end=' | ')
        c+=1
    m+=1

Enter the First Number : 10


2 | 6 | 12 | 20 | 30 | 42 | 56 | 72 | 90 | 110 | 

# **15. TWISTED PRIME NUMBER PROGRAM :**
### The Number itself should be Prime Number and Reverse of that number is also be a Prime Number
### Example : num = 31
### Check Prime Number
### 31%1==0 True - Count = 1
### 31%2==0 False 
### 31%3==0 False 
### 31%4==0 False 
### 31%5==0 False 
### 31%6==0 False 
...............
### 31%31==0 True - Count = 2
### If Cont == 2
### 2 == 2 True - Prime Number

### Reverse thr Number Check Prime Number
### 13%1==0 True - Count = 1
### 13%2==0 False 
### 13%3==0 False 
### 13%4==0 False 
### 13%5==0 False 
### 13%6==0 False 
..................
### 13%13==0 True - Count = 2
### If Cont == 2
### 2 == 2 True - Prime Number

### **15.1 CHECK WHETHER GIVEN NUMBER IS TWISTED PRIME NUMBER OR NOT**

In [2]:
num=int(input('Enter the Number'))
start=1
count=0
while start<=num:
    if num%start==0:
        count+=1
    start+=1
if count==2:
    rev=0
    while num!=0:
        d=num%10
        rev=rev*10+d
        num//=10
    count1=0
    start1=1    
    while start1<=rev:
        if rev%start1==0:
            count1+=1
        start1+=1    
    if count1==2:
        print('Twisted Prime Number')
    else:
        print('Not a Twisted Prime Number')
else:
    print('Not a Twisted Prime Number')

Enter the Number 97


Twisted Prime Number


### **15.2 PRINT ALL TWISTED PRIME NUMBERS FROM M TO N**

In [4]:
m=int(input('Enter the Start Number'))
n=int(input('Enter the End Number :'))
print('TWISTED PRIME NUMBER')
while m<=n:
    num=m
    start=1
    count=0
    while start<=num:
        if num%start==0:
            count+=1
        start+=1
    if count==2:
        rev=0
        while num!=0:
            d=num%10
            rev=rev*10+d
            num//=10
        count1=0
        start1=1    
        while start1<=rev:
            if rev%start1==0:
                count1+=1
            start1+=1    
        if count1==2:
            print(m,end=' | ')
    m+=1

Enter the Start Number 1
Enter the End Number : 100


TWISTED PRIME NUMBER
2 | 3 | 5 | 7 | 11 | 13 | 17 | 31 | 37 | 71 | 73 | 79 | 97 | 

### **15.3 PRINT FIRST N TWISTED PRIME NUMBERS**

In [3]:
m=1
n=int(input('Enter the First N Number :'))
print('TWISTED PRIME NUMBER')
c=0
while c<n:
    num=m
    start=1
    count=0
    while start<=num:
        if num%start==0:
            count+=1
        start+=1
    if count==2:
        rev=0
        while num!=0:
            d=num%10
            rev=rev*10+d
            num//=10
        count1=0
        start1=1    
        while start1<=rev:
            if rev%start1==0:
                count1+=1
            start1+=1    
        if count1==2:
            print(m,end=' | ')
            c+=1
    m+=1

Enter the First N Number : 10


TWISTED PRIME NUMBER
2 | 3 | 5 | 7 | 11 | 13 | 17 | 31 | 37 | 71 | 

# **16. MEGA PRIME NUMBER PROGRAM :**
### The Number itself should be Prime Number and Each Digit of that number is also be a Prime Number
### Example : num = 53
### Check Prime Number
### 53%1==0 True - Count = 1
### 53%2==0 False 
### 53%3==0 False 
### 53%4==0 False 
### 53%5==0 False 
### 53%6==0 False 
...............
### 53%53==0 True - Count = 2
### If Cont == 2
### 2 == 2 True - Prime Number

### Check Each Digit of the given Number is Prime Number 
### Num = 53
### First Digit = 5
### 5%1==0 True - Count = 1
### 5%2==0 False 
### 5%3==0 False 
### 5%4==0 False 
### 5%5==0 True - Count = 2
### If Cont == 2
### 2 == 2 True - Prime Number
### First Digit = 3
### 3%1==0 True - Count = 1
### 3%2==0 False 
### 3%3==0 True - Count = 2
### If Cont == 2
### 2 == 2 True - Prime Number

### **16.1 CHECK THE GIVEN NUMBER IS MEGA NUMBER OR NOT**

In [8]:
num=int(input('Enter the Number'))
start=1
count=0
while start<=num:
    if num%start==0:
        count+=1
    start+=1
if count==2:
    num1=num
    flag=0
    while num1!=0:
        d=num1%10
        start1=1
        count1=0
        while start1<=d:
            if d%start1==0:
                count1+=1
            start1+=1
        if count1==2:
            flag=1
            break
        num1//=10
    if flag==1:
        print('Mega Prime Number')
    else:
        print('Not a Mega Prime Number')
else:
    print('Not a Mega Prime Number')

Enter the Number 53


Mega Prime Number


### **16.2 PRINT ALL MEGA NUMBERS FROM M TO N**

In [9]:
m=int(input('Enter the Start Number :'))
n=int(input('Enter the End Number :'))
print('Mega Prime Number :')
while m<=n:
    num=m
    num1=num
    start=1
    count=0
    while start<=num:
        if num%start==0:
            count+=1
        start+=1
    flag=1
    if count==2:
        while num1!=0:
            d=num1%10
            start1=1
            count1=0
            while start1<=d:
                if d%start1==0:
                    count1+=1
                start1+=1
            if count1!=2:
                flag=0
                break
            num1//=10
        if flag==1:
            print(m,end=' | ')
    m+=1

Enter the Start Number : 1
Enter the End Number : 100


Mega Prime Number :
2 | 3 | 5 | 7 | 23 | 37 | 53 | 73 | 

### **16.3 PRINT FIRST N MEGA PRIME NUMBERS**

In [10]:
m=1
n=int(input('Enter the First N Number :'))
print('Mega Prime Number :')
c=0
while c<n:
    num=m
    num1=num
    start=1
    count=0
    while start<=num:
        if num%start==0:
            count+=1
        start+=1
    flag=1
    if count==2:
        while num1!=0:
            d=num1%10
            start1=1
            count1=0
            while start1<=d:
                if d%start1==0:
                    count1+=1
                start1+=1
            if count1!=2:
                flag=0
                break
            num1//=10
        if flag==1:
            print(m,end=' | ')
            c+=1
    m+=1

Enter the First N Number : 15


Mega Prime Number :
2 | 3 | 5 | 7 | 23 | 37 | 53 | 73 | 223 | 227 | 233 | 257 | 277 | 337 | 353 | 

# **17. SUNNY NUMBER PROGRAM :**
### **Sum** of Each Digit from given number == check that **Sum** is a Perfect Square 
### Example : num = 8
### Sum of Each digit
### Sum = 8 + 1 = 9
### Check Perfect Square
### start from 1
### 1*1 = sum False
### 2*2 = sum False
### 3*3 = sum True

### **17.1 CHECK GIVEN NUMBER IS SUNNY NUMBER OR NOT**

In [17]:
num=int(input())
is_sunny=False
for i in range(num+1):
    n=i*i
    if num+1==n:
        is_sunny=True
        break
if is_sunny:
    print('Sunny Number')
else:
    print('Not a Sunny Number')

 15


Sunny Number


#### OR

In [18]:
num=int(input())
i=0
flag=0
while i<=num+1:
    n=i*i
    if num+1==n:
        flag=1
        break
    i+=1
if flag:
    print('Sunny Number')
else:
    print('Not a Sunny Number')

 8


Sunny Number


### **17.2 PRINT ALL SUNNY NUMBERS FROM M TO N**

In [1]:
m=int(input('Enter the Start Number :'))
n=int(input('Enter the End Number :'))
print('SUNNY NUMBER')
while m<=n:
    num=m
    i=0
    flag=0
    while i<=num+1:
        sqr=i*i
        if num+1==sqr:
            flag=1
            break
        i+=1
    if flag==1:
        print(m,end=' | ')
    m+=1

Enter the Start Number : 1
Enter the End Number : 100


SUNNY NUMBER
3 | 8 | 15 | 24 | 35 | 48 | 63 | 80 | 99 | 

### **17.3 PRINT FIRST N SUNNY NUMBER**

In [1]:
m=1
n=int(input('Enter the First N Number :'))
c=0
print('SUNNY NUMBER')
while c<n:
    num=m
    i=0
    flag=0
    while i<=num+1:
        sqr=i*i
        if num+1==sqr:
            flag=1
            break
        i+=1
    if flag==1:
        print(m,end=' | ')
        c+=1
    m+=1

Enter the First N Number : 10


SUNNY NUMBER
3 | 8 | 15 | 24 | 35 | 48 | 63 | 80 | 99 | 120 | 

# **18. AUTOMORPHIC NUMBER PROGRAM :**
### **Square** of the given number == last digit of that squared number is eqauls to that number
### Example : num = 25
### Square = 25 ** 2 = 625
### Square of last digit == number
### 625 = 25
### check by each digit
### 5 == 5 True
### 2 == 2 True

### **18.1 CHECK GIVEN NUMBER IS AUTOMORPHIC NUMBER OR NOT**

In [13]:
num=int(input())
flag=1
sqr=num*num
while num!=0:
    ld1=num%10
    ld2=sqr%10
    if ld1!=ld2:
        flag=0
        break
    num//=10
    sqr//=10
if flag:
    print('Automorphic Number')
else:
    print('Not a Automorphic Number')

 25


Automorphic Number


#### OR

In [3]:
num=int(input())
is_automorphic=True
sqr=num*num
while num!=0:
    ld1=num%10
    ld2=sqr%10
    if ld1!=ld2:
        is_automorphic=False
        break
    num//=10
    sqr//=10
if is_automorphic:
    print('Automorphic Number')
else:
    print('Not a Automorphic Number')

 76


Automorphic Number


### **18.2 PRINT ALL AUTOMORPHIC NUMBER FROM M TO N**

In [2]:
m=int(input('Enter the Start Number :'))
n=int(input('Enter the End Number :'))
print('Automorphic Number :')
while m<=n:
    num=m
    flag=1
    sqr=num*num
    while num!=0:
        ld1=num%10
        ld2=sqr%10
        if ld1!=ld2:
            flag=0
            break
        num//=10
        sqr//=10
    if flag:
        print(m,end=' | ')
    m+=1

Enter the Start Number : 1
Enter the End Number : 100


Automorphic Number :
1 | 5 | 6 | 25 | 76 | 

### **18.3 PRINT FIRST N AUTOMORPHIC NUMBER**

In [4]:
m=1
n=int(input('Enter the First N Number :'))
print('Automorphic Number :')
c=0
while c<n:
    num=m
    flag=1
    sqr=num*num
    while num!=0:
        ld1=num%10
        ld2=sqr%10
        if ld1!=ld2:
            flag=0
            break
        num//=10
        sqr//=10
    if flag:
        print(m,end=' | ')
        c+=1
    m+=1

Enter the First N Number : 10


Automorphic Number :
1 | 5 | 6 | 25 | 76 | 376 | 625 | 9376 | 90625 | 109376 | 

# **19. MAGIC NUMBER PROGRAM :**
### **Sum** of the digit og the given number should be single digit and that is equals to the 1
### Example : num = 73
### 7 + 3 = 10 Single Digit = False
### 1 + 0 = 1 Single Digit = True
### 1 == 1

### **19.1 CHECK GIVEN NUMBER IS MAGIC NUMBER OR NOT**

In [12]:
num=int(input())
while num>9:
    sum=0
    while num!=0:
        d=num%10
        sum+=d
        num//=10
    num=sum
if num==1:
    print('Magic Number')
else:
    print('Not a Magic Number')

 73


Magic Number


### **19.2 PRINT ALL MAGIC NUMBERS FROM M TO N**

In [6]:
m=int(input('Enter the Start Number :'))
n=int(input('Enter the End Number :'))
print('MAGIC NUMBER')
while m<=n:
    num=m
    while num>9:
        sum=0
        while num!=0:
            d=num%10
            sum+=d
            num//=10
        num=sum
    if num==1:
        print(m,end=' | ')
    m+=1

Enter the Start Number : 1
Enter the End Number : 100


MAGIC NUMBER
1 | 10 | 19 | 28 | 37 | 46 | 55 | 64 | 73 | 82 | 91 | 100 | 

### **19.3 PRINT FIRST N MAGIC NUMBER**

In [7]:
m=1
n=int(input('Enter the First N Number :'))
c=0
print('MAGIC NUMBER')
while c<n:
    num=m
    while num>9:
        sum=0
        while num!=0:
            d=num%10
            sum+=d
            num//=10
        num=sum
    if num==1:
        print(m,end=' | ')
        c+=1
    m+=1

Enter the First N Number : 15


MAGIC NUMBER
1 | 10 | 19 | 28 | 37 | 46 | 55 | 64 | 73 | 82 | 91 | 100 | 109 | 118 | 127 | 

# **20. TECH NUMBER PROGRAM :**
### Count the Digit of a number and check that digit should be Even number, then split the number by 2 halves and make separate halves and add that halves and square after adding the halves that should be equals to the that number itself
### Example : num = 2025
### count of digit in number = 4
### 4%2==0 is even
### Split the number by two halves i.e, 2025 -> 20  |  25
### add the halves = 20 + 25 = 45
### Square the result = 45*45 = 2025
### 2025 == 2025(num)

### **20.1 CHECK THE GIVEN NUMBER IS TECH NUMBER OR NOT**

In [12]:
num=int(input())
count=0
num1=num
while num!=0: #count digits
    count+=1
    num//=10
if count%2==0: #check digit is even
    div=10**(count//2) #split by 2 halves
    fd=num1//div
    ld=num1%div
    res=fd+ld
    sqr=res*res #square the sum
    if sqr==num1:
        print('Tech Number')
    else:
        print('Not a Tech Number')
else:
    print('Not a Tech Number')

 2025


Tech Number


### **20.2 PRINT ALL TECH NUMBER FROM M TO N**

In [11]:
m=int(input('Enter the First Number :'))
n=int(input('Enter the End Number :'))
print('TECH NUMBERS')
while m<=n:
    num=m
    count=0
    num1=num
    while num!=0: #count digits
        count+=1
        num//=10
    if count%2==0: #check digit is even
        div=10**(count//2) #split by 2 halves
        fd=num1//div
        ld=num1%div
        res=fd+ld
        sqr=res*res #square the sum
        if sqr==num1:
            print(m,end=' | ')
    m+=1

Enter the First Number : 1
Enter the End Number : 3000


TECH NUMBERS
81 | 2025 | 

### **20.3 PRINT FIRST N TECH NUMBERS**

In [1]:
m=1
n=int(input('Enter the First N Number :'))
print('TECH NUMBERS')
c=0
while c<n:
    num=m
    count=0
    num1=num
    while num!=0: #count digits
        count+=1
        num//=10
    if count%2==0: #check digit is even
        div=10**(count//2) #split by 2 halves
        fd=num1//div
        ld=num1%div
        res=fd+ld
        sqr=res*res #square the sum
        if sqr==num1:
            print(m,end=' | ')
            c+=1
    m+=1

Enter the First N Number : 5


TECH NUMBERS
81 | 2025 | 3025 | 9801 | 494209 | 

# **21. ADAM NUMBER PROGRAM :**
### Revers of the square Number and reverse it again == square of that number
### Example : num = 12
### Square of that Number 
### 12*12=144
### Reverse the Number num =21
### Square the reverse Number 
### 21*21=441 if you revers it again both should be same 
### 144 == 441(reversed)
### 144 == 144

### **21.1 CHECK THAT GIVEN NUMBER IS ADAM NUMBER OR NOT**

In [11]:
num=int(input())
sqr=num**2
rev=0
while num!=0:
    d=num%10
    rev=rev*10+d
    num//=10
rev_sqr=rev**2
rev1=0
while rev_sqr!=0:
    d1=rev_sqr%10
    rev1=rev1*10+d1
    rev_sqr//=10
if sqr==rev1:
    print('Adam Number')
else:
    print('Not a Adam Number')

 12


Adam Number


### **21.2 PRINT ALL THE ADAM NUMBERS FROM M TO N**

In [8]:
m=int(input('Enter the Start Number :'))
n=int(input('Enter the End Number :'))
print('ADAM NUMBERS')
while m<=n:
    num=m
    sqr=num**2
    rev=0
    while num!=0:
        d=num%10
        rev=rev*10+d
        num//=10
    rev_sqr=rev**2
    rev1=0
    while rev_sqr!=0:
        d1=rev_sqr%10
        rev1=rev1*10+d1
        rev_sqr//=10
    if sqr==rev1:
        print(m,end=' | ')
    m+=1

Enter the Start Number : 1
Enter the End Number : 100


ADAM NUMBERS
1 | 2 | 3 | 11 | 12 | 13 | 21 | 22 | 31 | 

### **21.3 PRINT FIRST N ADAM NUMBERS**

In [9]:
m=1
n=int(input('Enter the First N Number :'))
print('ADAM NUMBERS')
c=0
while c<n:
    num=m
    sqr=num**2
    rev=0
    while num!=0:
        d=num%10
        rev=rev*10+d
        num//=10
    rev_sqr=rev**2
    rev1=0
    while rev_sqr!=0:
        d1=rev_sqr%10
        rev1=rev1*10+d1
        rev_sqr//=10
    if sqr==rev1:
        print(m,end=' | ')
        c+=1
    m+=1

Enter the First N Number : 15


ADAM NUMBERS
1 | 2 | 3 | 11 | 12 | 13 | 21 | 22 | 31 | 101 | 102 | 103 | 111 | 112 | 113 | 

# **22. PETERSON/KRISHNAMURTHY NUMBER PROGRAM :**
### Sum of factorial of digits == original number
### Example : num = 145
### 1!+4!+5! == 145
### 1+24+120 = 145
### 145 == 145

### **22.1 CHECK GIVEN NUMBER IS PETERSON/KRISHNAMURTHY NUMBER OR NOT**

In [13]:
num=int(input())
num1=num
sum=0
while num!=0:
    d=num%10
    start=1
    fact=1
    while start<=d:
        fact*=start
        start+=1
    sum+=fact
    num//=10
if num1==sum:
    print('Peterson or Krishnamurthy Number')
else:
    print('Not a Peterson or Krishnamurthy Number')

 145


Peterson or Krishnamurthy Number


### **22.2 PRINT ALL PETERSON OR KRISHNAMURTHY NUMBERS FROM M TO N**

In [17]:
m=int(input('Enter the Start Number :'))
n=int(input('Enter the End Number :'))
print('PETERSON OR KRISHNAMURTHY NUMBER')
while m<=n:
    num=m
    num1=num
    sum=0
    while num!=0:
        d=num%10
        start=1
        fact=1
        while start<=d:
            fact*=start
            start+=1
        sum+=fact
        num//=10
    if num1==sum:
        print(m,end=' | ')
    m+=1

Enter the Start Number : 1
Enter the End Number : 1000


PETERSON OR KRISHNAMURTHY NUMBER
1 | 2 | 145 | 

### **22.3 PRINT THE FIRST N PETERSON OR KRISHNAMURTHY NUMBER**

In [1]:
m=1
n=int(input('Enter the First N Number :'))
print('PETERSON OR KRISHNAMURTHY NUMBER')
c=0
while c<n:
    num=m
    num1=num
    sum=0
    while num!=0:
        d=num%10
        start=1
        fact=1
        while start<=d:
            fact*=start
            start+=1
        sum+=fact
        num//=10
    if num1==sum:
        print(m,end=' | ')
        c+=1
    m+=1

Enter the First N Number : 4


PETERSON OR KRISHNAMURTHY NUMBER
1 | 2 | 145 | 40585 | 

# **23. SMITH NUMBER PROGRAM :**
### Sum of digits of the number == Sum of digits of its prime factors
### Note : Number Must not be a Prime , we uses prime factors
### Example : num = 666
### Sum of Digits of a given number
### 6+6+6=18
### Find Prime Factorization
### strat=2 always because we are checking prime factors
### INCOMPLETE #***************************************************************************************************************#
### num2=666, start=2, num2%start, 666%2==0 True , num1=start, num1=2 , sum of that num1 is sum2 = 2 , num2 = 666//2==333 start+=1 -> start = 2+1 = 3
### num2=333, 333%3==0 True , num1=3 , sum of that num1 is sum2 = 3 , num2 = 333//3==111 start+=1 -> start = 3+1=4
### num2=111, 111%4==0 False , num1=4 , sum of that num1 is sum2 = 4 , num2 = 111//3==37 start+=1 -> start = 4+1=5
### num2=37, 37%5==0 False , start=5+1=6
### num2=37, 37%6==0 False , start=6+1=7
### num2=37, 37%7==0 False , start=7+1=8
### ....................................
### num2=37, 37%37==0 True , num1=37 , sum of that num1 is sum2 = 226 , num2 = 37//37==1 start+=1 -> start = 37+1=38

In [1]:
num=int(input())
num1=num
sum=0
# Find the sum of the digits :
while num1!=0:
    d=num1%10
    sum+=d
    num1//=10
# Find Prime Factorization :
sum2=0
num2=num
start=2
while start<=num2:
    while num2%start==0:
        num1=start
        while num1!=0:
            d2=num1%10
            sum2+=d2
            num1//=10
        num2//=start
    start+=1
if sum==sum2:
    print('Smith Number')
else:
    print('Not a Smith Number')

 666


Smith Number


# **24. BUZZ NUMBER PROGRAM :**
### Number should be either divisible by 7 or ends with 7
### Example = 17,27,37,77,1777777,100007
### num = 21
### 21%7==0 True

In [7]:
num=int(input())
flag=0
n=num
while num!=0:
    d=num%10
    if d==7:
        flag=1
        break
    else:
        if n%7==0:
            flag=1
            break
if flag:
    print('Buzz Number')
else:
    print('Not a Buzz Number')

 49


Buzz Number


# **25. EVIL NUMBER PROGRAM :**
### If count of 1s is even → **Evil Number**
### If count of 1s is odd → **Odious Number** (not evil)
### Example : num = 9
### Convert to binary
### | Division | Quotient | Remainder |
### | 9 ÷ 2    | 4        | 1         |
### | 4 ÷ 2    | 2        | 0         |
### | 2 ÷ 2    | 1        | 0         |
### | 1 ÷ 2    | 0        | 1         |
### Take Binary Numbers and check that the 1's number is even
### 1001 = 1's are = 2 , 2%2==0 True
### 9 is an EVIL NUMBER

In [9]:
num=int(input())
count=0
while num!=0:
    rem=num%2
    if rem==1:
        count+=1
    num//=2
if count%2==0:
    print("Evil Number")
else:
    print("Not an Evil Number")

 9


Evil Number


# **26. KAPREKAR NUMBER PROGRAM :**
### Square the number
### Split the square into two parts
### Add both parts
### If the result equals the original number, it is a Kaprekar Number

In [15]:
num=int(input())
sqr=num*num
num1=sqr
num2=num
count=0
while sqr!=0:
    count+=1
    sqr//=10
if count%2==0: 
    div=10**(count//2) 
    fd=num1//div
    ld=num1%div
    res=fd+ld
if res==num2:
    print('Kaprekar Number')
else:
    print('Not a Kaprekar Number')

 9


Kaprekar Number


# **27. ABUNDANT NUMBER PROGRAM :**
### Sum of factors (excluding the number)>number

In [16]:
num=int(input())
sum=0
start=1
while start<num:
    if num%start==0:
        sum+=start
    start+=1
if sum>num:
    print('Abundant Number')
else:
    print('Not a Abundant Number')

 12


Abundant Number


## ARMSTRONG NUMBER QUESTIONS
#### 1. Check if a number is Armstrong.
#### 2. Print all Armstrong numbers between 1–1000.
#### 3. Print Armstrong numbers between 1000–9999.
#### 4. Count Armstrong numbers in a range.
#### 5. Find largest Armstrong number below n.
#### 6. Check if reversed number is also Armstrong.
#### 7. Print first n Armstrong numbers.
#### 8. Find sum of all Armstrong numbers in range.
#### 9. Check Armstrong number without using string.
#### 10. Optimize Armstrong check for large numbers.

## STRONG NUMBER QUESTIONS
#### 11. Check Strong number.
#### 12. Print Strong numbers between 1–1000.
#### 13. Count Strong numbers in range.
#### 14. Check if reversed number is Strong.
#### 15. Print first n Strong numbers.
#### 16. Find sum of Strong numbers in range.
#### 17. Check if sum of digits factorial is prime.
#### 18. Check Strong number without using math module.
#### 19. Optimize factorial calculation.
#### 20. Find largest Strong number below n.

## PERFECT NUMBER QUESTIONS
#### 21. Check Perfect number.
#### 22. Print Perfect numbers between 1–10,000.
#### 23. Count Perfect numbers in range.
#### 24. Print first n Perfect numbers.
#### 25. Check if sum of divisors is prime.
#### 26. Find difference between Perfect and Abundant.
#### 27. Check Perfect using optimized divisor logic.
#### 28. Find largest Perfect below n.
#### 29. Print Perfect numbers in given list.
#### 30. Count even Perfect numbers in range.

## HARSHAD NUMBER QUESTIONS
#### 31. Check Harshad number.
#### 32. Print Harshad numbers between 1–1000.
#### 33. Count Harshad numbers in range.
#### 34. Check if reversed number is Harshad.
#### 35. Print first n Harshad numbers.
#### 36. Find largest Harshad below n.
#### 37. Count even Harshad numbers.
#### 38. Find Harshad numbers divisible by 3.
#### 39. Check if Harshad remains Harshad after division.
#### 40. Optimize Harshad logic.

## HAPPY NUMBER QUESTIONS
#### 41. Check Happy number.
#### 42. Print Happy numbers between 1–100.
#### 43. Count Happy numbers in range.
#### 44. Find next Happy number after n.
#### 45. Print first n Happy numbers.
#### 46. Check if reversed number is Happy.
#### 47. Detect cycle without using set.
#### 48. Count steps needed to reach 1.
#### 49. Find smallest Happy number greater than 100.
#### 50. Check if number becomes Happy after adding 1.

## NEON NUMBER QUESTIONS
#### 51. Check Neon number.
#### 52. Print Neon numbers between 1–1000.
#### 53. Count Neon numbers in range.
#### 54. Find sum of Neon numbers.
#### 55. Check if reversed number is Neon.
#### 56. Print first n Neon numbers.
#### 57. Check Neon without converting to string.
#### 58. Find largest Neon below n.
#### 59. Count even Neon numbers.
#### 60. Combine Neon and Harshad check.

## AUTOMORPHIC NUMBER QUESTIONS
#### 61. Check Automorphic number.
#### 62. Print Automorphic numbers in range.
#### 63. Count Automorphic numbers.
#### 64. Check reversed number Automorphic.
#### 65. Print first n Automorphic numbers.
#### 66. Find largest Automorphic below n.
#### 67. Count even Automorphic numbers.
#### 68. Check Automorphic without string method.
#### 69. Combine Automorphic & Neon check.
#### 70. Find square length dynamically.

## SPY NUMBER QUESTIONS
#### 71. Check Spy number.
#### 72. Print Spy numbers between 1–1000.
#### 73. Count Spy numbers.
#### 74. Check reversed number Spy.
#### 75. Print first n Spy numbers.
#### 76. Find largest Spy below n.
#### 77. Count even Spy numbers.
#### 78. Combine Spy & Harshad check.
#### 79. Find Spy numbers divisible by 5.
#### 80. Optimize Spy logic.

## KAPREKAR NUMBER QUESTIONS
#### 81. Check Kaprekar number.
#### 82. Print Kaprekar numbers in range.
#### 83. Count Kaprekar numbers.
#### 84. Find next Kaprekar number.
#### 85. Print first n Kaprekar numbers.
#### 86. Check reversed number Kaprekar.
#### 87. Find largest Kaprekar below n.
#### 88. Count even Kaprekar numbers.
#### 89. Combine Kaprekar & Perfect check.
#### 90. Optimize split logic.

## ABUNDANT NUMBER QUESTIONS
#### 91. Check Abundant number.
#### 92. Print Abundant numbers between 1–1000.
#### 93. Count Abundant numbers.
#### 94. Print first n Abundant numbers.
#### 95. Find smallest Abundant number.
#### 96. Check reversed number Abundant.
#### 97. Count even Abundant numbers.
#### 98. Combine Abundant & Harshad.
#### 99. Find difference between Abundant and Perfect.
#### 100. Optimize divisor logic.